File: apps/old-api/src/main/java/pt/iade/ei/thinktoilet/models/requests/CommentRequest.java
package pt.iade.ei.thinktoilet.models.requests;

import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CommentRequest {
    private int toiletId;
    private int userId;
    @Size(max = 280)
    private String text;
    private int ratingClean;
    private boolean ratingPaper;
    private int ratingStructure;
    private int ratingAccessibility;
}

File: app/app/src/main/java/pt/iade/ei/thinktoilet/models/requests/CommentRequest.kt
package pt.iade.ei.thinktoilet.models.requests

import com.google.gson.annotations.SerializedName

data class CommentRequest(
    @SerializedName("toiletId") val toiletId: Int,
    @SerializedName("userId") val userId: Int,
    @SerializedName("text") val text: String,
    @SerializedName("ratingClean") val ratingClean: Int,
    @SerializedName("ratingPaper") val ratingPaper: Boolean,
    @SerializedName("ratingStructure") val ratingStructure: Int,
    @SerializedName("ratingAccessibility") val ratingAccessibility: Int
)
